xds/src/main/java/io/grpc/xds/ClusterManagerLoadBalancer.java

```diff
@@ -0,0 +1,264 @@
+        }
+      };
+      helper.updateBalancingState(overallState, picker);
+    }
+  }
+
+  @Nullable
+  private static ConnectivityState aggregateState(
+      @Nullable ConnectivityState overallState, ConnectivityState childState) {
+    if (overallState == null) {
+      return childState;
+    }
+    if (overallState == READY || childState == READY) {
+      return READY;
+    }
+    if (overallState == CONNECTING || childState == CONNECTING) {
+      return CONNECTING;
+    }
+    if (overallState == IDLE || childState == IDLE) {
+      return IDLE;
+    }
+    return overallState;
+  }
+
+  private final class ChildLbState {
+    private final String name;
+    private final GracefulSwitchLoadBalancer lb;
+    private LoadBalancerProvider policyProvider;
+    private ConnectivityState currentState = CONNECTING;
+    private SubchannelPicker currentPicker = BUFFER_PICKER;
+    private boolean deactivated;
+    @Nullable
+    ScheduledHandle deletionTimer;
+
+    ChildLbState(String name, LoadBalancerProvider policyProvider) {
+      this.name = name;
+      this.policyProvider = policyProvider;
+      lb = new GracefulSwitchLoadBalancer(new ChildLbStateHelper());
+      lb.switchTo(policyProvider);
+    }
+
+    void deactivate() {
+      if (deactivated) {
+        return;
+      }
+
+      class DeletionTask implements Runnable {
+        @Override
+        public void run() {
+          shutdown();
+          childLbStates.remove(name);
+        }
+      }
+
+      deletionTimer =
+          syncContext.schedule(
+              new DeletionTask(),
+              DELAYED_ACTION_DELETION_TIME_MINUTES,
+              TimeUnit.MINUTES,
+              timeService);
+      deactivated = true;
+      logger.log(XdsLogLevel.DEBUG, "Child balancer {0} deactivated", name);
+    }
+
+    void reactivate(LoadBalancerProvider policyProvider) {
+      if (deletionTimer != null && deletionTimer.isPending()) {
+        deletionTimer.cancel();
+        deactivated = false;
+        logger.log(XdsLogLevel.DEBUG, "Child balancer {0} reactivated", name);
+      }
+      if (!this.policyProvider.getPolicyName().equals(policyProvider.getPolicyName())) {
+        logger.log(
+            XdsLogLevel.DEBUG,
+            "Child balancer {0} switching policy from {1} to {2}",
+            name, this.policyProvider.getPolicyName(), policyProvider.getPolicyName());
+        lb.switchTo(policyProvider);
+        this.policyProvider = policyProvider;
+      }
+    }
+
+    void shutdown() {
+      deactivated = true;
+      if (deletionTimer != null && deletionTimer.isPending()) {
+        deletionTimer.cancel();
+      }
+      lb.shutdown();
+      logger.log(XdsLogLevel.DEBUG, "Child balancer {0} deleted", name);
+    }
+
+    private final class ChildLbStateHelper extends ForwardingLoadBalancerHelper {
+
+      @Override
+      public void updateBalancingState(ConnectivityState newState, SubchannelPicker newPicker) {
+        if (deactivated) {
```

Comment:
Even if deactivated, should still update `currentState` and `currentPicker`, just not `updateOverallBalancingState()` (already guaranteed). The `XdsRoutingLoadBalancer` has the same issue.

Reply:
Good point. Fixed here.